core/src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -2919,4 +2919,40 @@ abstract class MergeIntoSuiteBase
     customConditionErrorRegex =
       Option("Aggregate functions are not supported in the .* condition of MERGE operation.*")
   )
+
+  Seq(true, false).foreach { differentActiveSession =>
```

Comment:
nit: the case where the session is the same doesn't bring a lot of value as it's widely covered by other tests already, you can remove it and simplify the test.